src/Strategy/Puli.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Http\Discovery\Strategy;
+
+use Http\Discovery\Exception\PuliUnavailableException;
+
+/**
+ * @author David de Boer <david@ddeboer.nl>
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+class Puli implements DiscoveryStrategy
+{
+    /**
+     * @var \Puli\GeneratedPuliFactory
+     */
+    private static $puliFactory;
+
+    /**
+     * @var \Puli\Discovery\Api\Discovery
```

Comment:
i think my IDE also complains about missing classes in annotations...
for the language php, its ok to have use statements for things that
don't exist. its even ok to have typehints as long as you default to
null and only pass null...

i'd prefer the use, to be consistent.